packages/migration/src/transforms/unit-tests/helpers.ts

```diff
@@ -0,0 +1,151 @@
+  isNockVariableDeclarationInstantiation, NockVariableDeclarationInstantiation,
+} from './validators.js'
+import {nullTransformationContext} from '../../nullTransformationContext.js'
+import _ from 'lodash'
+
+const {factory} = ts
+
+export type NockIntercepts = { varName: string, domain: ts.StringLiteral, intercepts: ts.CallExpression[] }
+// top level keys are varNames used when defining nock instance
+export type NockInterceptsLookup = Record<string, NockIntercepts>
+
+const stripNockFromCallAccessChain = (callEx: ts.CallExpression, varName: string): ts.CallExpression => {
+  const visitor = (node: ts.Node): ts.Node => {
+    if (
+      ts.isCallExpression(node) &&
+      ts.isIdentifier(node.expression) &&
+      node.expression.escapedText.toString() === 'nock'
+    ) {
+      return factory.createIdentifier(varName)
+    }
+
+    return ts.visitEachChild(node, visitor, nullTransformationContext)
+  }
+
+  return ts.visitEachChild(callEx, visitor, nullTransformationContext)
+}
+
+/* nodes are "upside down" compared to human reading. Given:
+ * let api = nock('https://api.heroku.com:443')
+   .post('/account/keys', {public_key: key})
+   .reply(200)
+ * node starts as `.reply(200)`, and last CallExpression/PropertyAccessExpression pair is our target `nock('https://api.heroku.com:443')`
+ * */
+export const getEndOfCallPropertyAccessChain = (node: ts.CallExpression): ts.Node => {
+  let workingNode: ts.Node = node
+
+  while (ts.isCallExpression(workingNode) && ts.isPropertyAccessExpression(workingNode.expression)) {
+    workingNode = workingNode.expression.expression
+  }
+
+  return workingNode
+}
+
+export const createNockNameCallPairFromVarDeclInstantiation = (statement: NockVariableDeclarationInstantiation) => {
+  const varName = statement.declarationList.declarations[0].name.escapedText.toString()
+  const nockCallEx = statement.declarationList.declarations[0].initializer
+  const nockCallTerminus = getEndOfCallPropertyAccessChain(nockCallEx)
+  if (ts.isCallExpression(nockCallTerminus) && ts.isStringLiteral(nockCallTerminus.arguments[0])) {
+    return {
+      varName,
+      domain: nockCallTerminus.arguments[0],
+      intercepts: [stripNockFromCallAccessChain(nockCallEx, varName)],
+    }
+  }
+
+  // this should never happen as the `if` of this `else` is mostly to soothe TS
+  throw new Error('createNockNameCallPairFromVarDeclInstantiation: found nock with unexpected shape')
+}
+
+export const getNockCallsFromBlock = (block: ts.Block, nestedNockInBeforeEach: NockInterceptsLookup) => {
+  // clone deeply here to avoid modifying anything that is passed by reference from separate node branches
+  nestedNockInBeforeEach = _.cloneDeep<NockInterceptsLookup>(nestedNockInBeforeEach)
+  for (const statement of block.statements) {
+    if (isNockExpressionStatementInstantiation(statement)) {
+      const varName = statement.expression.left.escapedText.toString()
+      if (nestedNockInBeforeEach[varName]) {
+        debugger
```

Comment:
Are you planning to remove these later on?

Reply:
I'll remove now. Good catch.